subsys/net/lib/tftp/tftp_client.c

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (c) 2020 InnBlue
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(tftp_client, LOG_LEVEL_DBG);
+
+/* Public / Internal TFTP header file. */
+#include <net/tftp.h>
+#include "tftp_client.h"
+
+/* TFTP Global Buffer. */
+static u8_t   tftpc_buffer[TFTPC_MAX_BUF_SIZE];
+static u32_t  tftpc_buffer_size;
+
+/* TFTP Request block number property. */
+static u32_t  tftpc_block_no;
```

Comment:
How presence of these global variables blends with:

1. Availability of `struct tftpc`.
2. Thus, apparent ability to have multiple `struct tftpc` in action.
3. Which can be executed concurrently in multiple threads.

?

Reply:
All of these are correlated to the global "tftpc_buffer" and will go away once the tftp client implementation starts using the user supplied buffer in a better way..